theano/gpuarray/opt.py

```diff
@@ -845,10 +848,14 @@ def local_gpu_pdbbreakpoint_op(node):
 def local_gpua_lazy_ifelse(op, context_name, inputs):
     if op.gpu:
         return
+    if isinstance(inputs[0].type, GpuArrayType):
+        return
     c = inputs[0]
     inps = []
     for v in inputs[1:]:
-        if isinstance(v.type, (tensor.TensorType, GpuArrayType)):
+        if isinstance(v.type, GpuArrayType):
+            return
+        elif isinstance(v.type, tensor.TensorType):
```

Comment:
Yeah, the output is also being checked, that is why that condition when we're strict. But one small mistake should be corrected. ```any()``` would return True if either one of them is True, so the previous Op has been transferred, but not the current one. That needs slight modification, correct ? https://github.com/Sentient07/Theano/blob/1736a6ce4ae50c6b3a6b4c286e1f23b8a43a0f33/theano/gpuarray/opt.py#L316